driver/compenv.ml

```diff
@@ -27,17 +29,19 @@ let print_version_and_library compiler =
   print_string Config.version; print_newline();
   print_string "Standard library directory: ";
   print_string Config.standard_library; print_newline();
-  exit 0
+  raise (Exit_compiler 0)
```

Comment:
Have you checked that there are no more calls to `exit` in the compiler code?

Reply:
Yes, I've changed all the ones I could find that might be reachable from ocamlc or ocamlopt. (There are others used by tools other than the compiler)